rest-service/manager_rest/rest/resources_v3_1/agents.py

```diff
@@ -27,32 +26,38 @@
 from manager_rest.rest.responses_v3 import AgentResponse
 from manager_rest.security.authorization import authorize
 from manager_rest.storage import models, get_storage_manager
-from manager_rest.resource_manager import get_resource_manager
 from manager_rest.rest.rest_utils import (validate_inputs,
-                                          get_json_and_verify_params,
-                                          get_args_and_verify_arguments)
+                                          verify_and_convert_bool,
+                                          get_json_and_verify_params)
 
 
 class Agents(SecuredResource):
     @rest_decorators.exceptions_handled
     @rest_decorators.marshal_with(AgentResponse)
+    @rest_decorators.create_filters(models.Agent)
     @rest_decorators.paginate
+    @rest_decorators.sortable(models.Agent)
+    @rest_decorators.all_tenants
+    @rest_decorators.search('name')
     @authorize('agent_list')
-    def get(self, pagination=None):
-        args = get_args_and_verify_arguments([
-            Argument('deployment_id', required=False),
-            Argument('node_ids', required=False, action='append'),
-            Argument('node_instance_ids', required=False,
-                     action='append'),
-            Argument('install_methods', required=False,
-                     action='append'),
-
-        ])
-        return get_resource_manager().list_agents(
-            deployment_id=args.get('deployment_id'),
-            node_ids=args.get('node_ids'),
-            node_instance_ids=args.get('node_instance_ids'),
-            install_method=args.get('install_methods'))
+    def get(self, _include=None, filters=None, pagination=None, sort=None,
+            all_tenants=None, search=None):
+
+        get_all_results = verify_and_convert_bool(
+            '_get_all_results',
+            request.args.get('_get_all_results', False)
+        )
+        agents_list = get_storage_manager().list(
```

Comment:
We are adding all this new functionality with filters, sort, include...
We should do one basic test for each one to make sure it works.
You can add the tests to `manager_rest/test/endpoints/test_agents.py`.
Also check the the old api requests works.